### ISSUE TITLE
Is gen_diffusive_1 interactive process to x0?

### ISSUE THREAD

**Issue by chengyu89527** [icon-lab/SynDiff#15]
When I check code. It said that the gen_diffusive_1 it gives the x0 ,but you define gen_diffusive_1 = NCSNpp(args).to(device). I can't find iterative operator during training. It seems that you using NSCNPP model generate the final result with one step ?

**Comment by onat-dalmaz** [comment]
https://github.com/icon-lab/SynDiff/blob/13dc0786b3d0f29e61486fb48d0021534bf22871/train.py#L587

As you can see, diffusive generator parameters are updated based on the calculated loss function.

**Comment by chengyu89527** [comment]
thanks for answer. However I check the Line 537 errG1_L1 = F.l1_loss(x1_0_predict_diff[:,[0],:],real_data1) . it measure the similarity of original image and generated x0. But the x1_0_predict_diff is from Line 543 x1_0_predict_diff = gen_diffusive_1(torch.cat((x1_tp1.detach(),x2_0_predict),axis=1), t1, latent_z1). it given by generator without diffusion step. I wonder if it's same with paper method?
